core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala

```diff
@@ -299,10 +313,22 @@ import scala.compat.java8.FutureConverters._
       override protected def onTimer(timerKey: Any): Unit = timerKey match {
         case QueryPoll | TagNotification =>
           continue()
+        case PersistenceIdsCleanup =>
+          cleanup()
       }
 
-      override def onPull(): Unit =
+      override def onPull(): Unit = {
         tryPushOne()
+      }
+
+      private def cleanup(): Unit = {
+        val now = System.currentTimeMillis()
+        val garbageCollected = stageState.tagPidSequenceNrs.filterNot {
```

Comment:
`garbageCollected` sounds like it contains the ones that are to be discarded but it's the remaining